### PR TITLE
Login when submitting OTP

### PR DESCRIPTION
**Pull request by vinaysshenoy**

### CODE REVIEW

**Comment by saket** [comment]
TODO: Get the design reviewed by Akshay tomorrow.

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreen.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package org.simple.clinic.enterotp
 
 import android.content.Context
+import android.support.v7.widget.Toolbar
 import android.util.AttributeSet
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreen.kt]: Stale commit?

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt [hunk 2]
```diff
@@ -12,19 +15,41 @@ import javax.inject.Inject
...
   }
 
+  private fun showOtpValidationErrors(events: Observable<UiEvent>): Observable<UiChange> {
+    val showIncorrectOtpErrors = events.ofType<EnterOtpSubmitted>()
+        .map { it.otp.length }
+        .map {
+          if (it != OTP_LENGTH) {
+            { ui: Ui -> ui.showIncorrectOtpError() }
+          } else {
+            { _ -> }
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt]: If you don't want to do anything, can we just use a filter here? :p

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt [hunk 3]
```diff
@@ -35,4 +60,29 @@ class EnterOtpScreenController @Inject constructor(
...
+            is LoginResult.ServerError -> { ui: Ui -> ui.showServerError(loginResult.error) }
+            is LoginResult.NetworkError -> { ui: Ui -> ui.showNetworkError() }
+          }
+        }
+  }
+
+  private fun startSyncing() {
+    syncScheduler.syncImmediately()
+        .subscribeOn(Schedulers.io())
+        // Swallow errors
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt]: This comment isn't necessary. `onErrorComplete()` already says the same thing :)

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreen.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreen.kt]: Whoops, yes. Happened while rerolling `master`. Will fix.

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt]: Right, sometimes my brain doesn't function.

File: app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt]: This was changed in the next commit. :P

**Comment by vinaysshenoy** [comment]
@saket I am planning to answer the `TODO` questions you have raised in my next PR that finishes up this feature.

**Pull request merged**